components/mediation-connector/org.wso2.carbon.connector.core/src/main/java/org/wso2/carbon/connector/core/RefreshAccessToken.java

```diff
@@ -0,0 +1,120 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.connector.core;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseLog;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.Set;
+
+public class RefreshAccessToken extends AbstractConnector {
+    private CloseableHttpClient httpClient = HttpClients.createDefault();
+
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+        SynapseLog synLog = getLog(messageContext);
+        Set propertyKeySet = messageContext.getPropertyKeySet();
+        propertyKeySet.remove("Accept-Encoding");
+
+
+        if (synLog.isTraceOrDebugEnabled()) {
+            synLog.traceOrDebug("Start : Salesforce Refresh Access Token mediator. If you want to observe refresh " +
+                    "access token header and wire logs please enable Apache HTTP Client logs in Log4j.properties");
+
+            if (synLog.isTraceTraceEnabled()) {
+                synLog.traceTrace("Message : " + messageContext.getEnvelope());
+            }
+        }
+
+        String refreshUrl;
+        String customRefreshUrl = (String) messageContext.getProperty("uri.var.customRefreshUrl");
+
+        if (customRefreshUrl != null && !StringUtils.isEmpty(customRefreshUrl)) {
```

Comment:
StringUtils.isEmpty does check for null as well. So we can remove the null check here